app/helpers/buyers/applications_helper.rb

```diff
@@ -1,49 +1,103 @@
-module Buyers::ApplicationsHelper
-
-  def metadata_new_app(buyer, provider)
+# frozen_string_literal: true
 
-    "<div id='metadata-form'
-      data-services_contracted='#{ services_contracted(buyer) }'
-      data-service_plan_contracted_for_service='#{ service_plan_contracted_for_service(buyer) }'
-      data-relation_service_and_service_plans='#{ relation_service_and_service_plans(provider) }'
-      data-create_service_plan_path='#{ create_service_plan_path }'
-      data-relation_plans_services= '#{ relation_plans_services(provider) }' >".html_safe
+module Buyers::ApplicationsHelper
 
+  def new_application_form_metadata(provider:, buyer: nil, service: nil, cinstance: nil)
+    {
+      'create-application-path': buyer ? admin_buyers_account_applications_path(buyer) : admin_buyers_applications_path,
+      'create-application-plan-path': new_admin_service_application_plan_path(':id'),
+      'create-service-plan-path': new_admin_service_service_plan_path(':id'),
+      'service-subscriptions-path': admin_buyers_account_service_contracts_path(':id'),
+      'service-plans-allowed': provider.settings.service_plans.allowed?.to_json,
+      product: service && product_data(service).to_json,
+      products: !service && data_products(provider).to_json,
+      buyer: buyer && buyer_data(buyer).to_json,
+      buyers: !buyer && data_buyers(provider).to_json,
+      'defined-fields': data_defined_fields(provider).to_json,
```

Comment:
Yes it's either this or to call `as_json` on every attribute and then `to_json` on the hash.

Reply:
You can do as_json on the general Hash, it will recursively call `as_json` to all the values

```rub
# activesupport-5.0.7.2/lib/active_support/core_ext/object/json.rb:147:
# Owner: Hash
# Visibility: public
# Signature: as_json(options=?)
# Number of lines: 16

def as_json(options = nil) #:nodoc:
  # create a subset of the hash by applying :only or :except
  subset = if options
    if attrs = options[:only]
      slice(*Array(attrs))
    elsif attrs = options[:except]
      except(*Array(attrs))
    else
      self
    end
  else
    self
  end

  Hash[subset.map { |k, v| [k.to_s, options ? v.as_json(options.dup) : v.as_json] }]
end
```

Reply:
Nevermind, it has to be done this way in order to have the correct attribute names (kebab-case to camelCase transformation).